third_party/amd/lib/TritonAMDGPUToLLVM/ElementwiseOpToLLVM.cpp

```diff
@@ -1134,12 +950,26 @@ struct FpToFpOpConversion
     }
     if (useFP16IntermediateSrc)
       for (Value &v : inVals)
-        v = convertFp32ToFp16NZ(loc, rewriter, v);
+        v = cvtFp32ToFp16(loc, rewriter, v,
+                          roundingMode.value_or(RoundingMode::RTNE));
     inVals.resize(numElements, undef(typeConverter->convertType(srcType)));
     SmallVector<Value> outVals;
     if (srcType != dstType) {
-      auto cvtFunc = getConversionFunc(srcType, dstType);
-      outVals = cvtFunc(loc, rewriter, inVals);
+      auto getCvtFunc = getConversionFunc(srcType, dstType, roundingMode);
+      if (failed(getCvtFunc)) {
+        std::string err_msg;
```

Comment:
I think there is no need to explicitly construct the stream. You can do `mlir::emitError(loc, "unsupported converstion from") << srcType << " to " << dstType << " with rounding mode " << stringifyRoundingMode(roundingMode.value_or(<default>)`?

Reply:
I didn't know we could do it this way. This is better. Changed.